doc/cookbook/engine.rst

```diff
@@ -0,0 +1,84 @@
+Engines
```

Comment:
```suggestion
Notebook executors
```

(let's also rename the file to nb_executors.rst)

Reply:
sorry for the naming changes. I realized that "engines" is a confusing name. it's better to call this "executors" to be consistent with the "NotebookExecutor" name.

so let's rename from `engine` -> `executor`, and `nb_executor_params` -> `executor_params`